pkg/crc/machine/bundle/metadata.go

```diff
@@ -346,7 +348,12 @@ func Download(preset crcPreset.Preset, bundleURI string) (string, error) {
 	if bundleURI == constants.GetDefaultBundlePath(preset) {
 		switch preset {
 		case crcPreset.OpenShift, crcPreset.Microshift:
-			return downloadDefault(preset)
+			downloadedBundlePath, err := downloadDefault(preset)
+			if grab.IsStatusCodeError(errors.Unwrap(err)) && enableBundleQuayFallback {
```

Comment:
Looks like this could be `errors.Is(err, grab.StatusCodeError)`

Reply:
@cfergeau this is not work because as per `errors.Is` it doesn't call the `unwrap` on the target and the way we have this error it is wrapped around `grabStatusCodeError` so it is not able to work.

Reply:
`errors.Is` was introduced to deal with wrapped errors. Its documentation is clear about this. https://pkg.go.dev/errors

Reply:
Yes but if I take a look to grab error package the implementation around StatusCodeError is type int and if I try to use `errors.Is` on that it is not working until we know the statuscode. I think that is the reason it have `IsStatusCodeError` function.

https://go.dev/play/p/jwkn9TG9SmH

Reply:
Ah right, `errors.Is` checks for equality, not for type :-/ `errors.As` checks for types, but is not as convenient as `errors.Is`. If we keep this code, I think we should use `errors.As` instead of hardcoding a single `errors.Unwrap()`. Hardcoding this `Unwrap()` will break if we move/refactor this code, and the level of wrapping is changed (eg if `errors.Unwrap(errors.Unwrap())` becomes needed instead of `errors.Unwrap()`). 
However looking at `downloadDefault()`, there are more situations than "the bundle file is missing/cannot be downloaded" when we want to fallback, for example if the signature is missing/invalid. `IsStatusCodeError()` seems to be too restrictive?

Reply:
@cfergeau yeah that is more flexible, I can revert to that.

Reply:
Done